### PR TITLE
Place picker tests

### PR DESCRIPTION
**Pull request by kejmil01**
TODO:
- edit activity tests

### CODE REVIEW

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisherTest.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    public void whenGoogleApiIsNotBuild_ApiIsNotConnected() {
+        //when
+        boolean apiConnected = establisher.isApiConnected();
+
+        //then
+        assertEquals(false, apiConnected);
+    }
+
+    @Test
+    public void whereConnectionIsRequestedAndApiClientIsBuild_GoogleApiConnectIsCalled() {
```
Comment (BartSoftware) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisherTest.java]: when*

File: app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisherTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+    @Before
+    public void setUp() {
+        Context context = RuntimeEnvironment.application;
+        GoogleApiClient.Builder builder = new GoogleApiClient.Builder(context);
+        establisher = new GoogleLocationApiEstablisher(builder);
+    }
+
+    @Test
+    public void whenGoogleApiIsNotBuild_ApiIsNotConnected() {
+        //when
```
Comment (BartSoftware) [inline app/src/test/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisherTest.java]: When, Then, Given (everywhere)

**Review by BartSoftware — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java [hunk 3]
```diff
@@ -1,17 +1,21 @@
...
 public class GoogleLocationApiEstablisher {
-    private GoogleApiConnectionListener googleApiConnectionListener;
+
     private GoogleApiClient.Builder googleApiClientBuilder;
     @Nullable
-    private GoogleApiClient googleApiClient;
+    private GoogleApiConnectionListener googleApiConnectionListener;
+    @Nullable
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    protected GoogleApiClient googleApiClient;
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java]: package protected

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java [hunk 4]
```diff
@@ -31,27 +35,37 @@ void requestConnection() {
...
-
-                    }
-                })
+                .addOnConnectionFailedListener(buildConnectionFailedListener())
+                .addConnectionCallbacks(buildConnectionCallbacks())
                 .build();
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    protected GoogleApiClient.OnConnectionFailedListener buildConnectionFailedListener() {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java]: package protected

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java [hunk 5]
```diff
@@ -31,27 +35,37 @@ void requestConnection() {
...
+        return connectionResult -> {
+            if (googleApiConnectionListener != null) {
+                googleApiConnectionListener.onFailure();
+            }
+        };
+    }
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    protected GoogleApiClient.ConnectionCallbacks buildConnectionCallbacks() {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java]: package protected

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java [hunk 6]
```diff
@@ -21,7 +22,8 @@
 public class GpsEstablisher {
     private static final String TAG = GpsEstablisher.class.getSimpleName();
 
-    private static final int GPS_REQUEST_CODE_DEFAULT = 1234;
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    protected static final int GPS_REQUEST_CODE_DEFAULT = 1234;
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: package protected

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java [hunk 7]
```diff
@@ -96,6 +91,19 @@ private void requestGps() {
         });
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    protected PendingResult<LocationSettingsResult> buildLocationSettingsResult() {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: package protected

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java [hunk 8]
```diff
@@ -126,6 +132,13 @@ private void onLocationResolutionRequired(Status status) {
         }
     }
 
+    @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
+    protected void startResolutionForResult(Status status) throws IntentSender.SendIntentException {
```
Comment (BartSoftware) [inline app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GpsEstablisher.java]: package protected

**Review by BartSoftware — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java (hunk 3 above)
Comment (kejmil01) [reply app/src/main/java/com/invisibleteam/goinvisible/mvvm/edition/GoogleLocationApiEstablisher.java]: Could you explain this request?

**Review by michal1990 — APPROVED** [review]

**Review by BartSoftware — APPROVED** [review]

**Pull request merged**